plugins/modules/nd_pcv.py

```diff
@@ -0,0 +1,211 @@
+    type: str
+  manual:
+    description:
+    - Optional parameter if creating new pre-change analysis from change-list (manual)
+    type: str
+  state:
+    description:
+    - Use C(query) for retrieving the version object.
+    type: str
+    choices: [ query ]
+    default: query
+extends_documentation_fragment: cisco.nd.modules
+'''
+
+EXAMPLES = r'''
+- name: Get prechange validation result
+  cisco.nd.nd_pcv:
+    insights_group: exampleIG
+    state: query
+  register: query_results
+- name: Get a specific prechange validation result
+  cisco.nd.nd_pcv:
+    insights_group: exampleIG
+    site_name: siteName
+    name: demoName
+    state: query
+  register: query_result
+- name: Create a new Pre-Change analysis from file
+  cisco.nd.nd_pcv:
+    insights_group: igName
+    site_name: siteName
+    name: demoName
+    file: configFilePath
+    state: present
+- name: Present Pre-Change analysis from manual changes
+  cisco.nd.nd_pcv:
+    insights_group: idName
+    site_name: SiteName
+    name: demoName
+    manual: |
+        [
+            {
+              "fvTenant": {
+                "attributes": {
+                  "name": "AnsibleTest",
+                  "dn": "uni/tn-AnsibleTest",
+                  "status": "deleted"
+                }
+              }
+            }
+        ]
+    state: present
+  register: present_pcv_manual
+- name: Delete Pre-Change analysis
+  cisco.nd.nd_pcv:
+    insights_group: igName
+    site_name: siteName
+    name: demoName
+    state: absent
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+from ansible_collections.cisco.nd.plugins.module_utils.ndi import NDI
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        insights_group=dict(type='str', required=True, aliases=[ "fab_name", "ig_name" ]),
+        name=dict(type='str'),
+        description=dict(type='str', aliases=[ "descr" ]),
+        site_name=dict(type='str', aliases=[ "site" ]),
+        file=dict(type='str'),
+        manual=dict(type='str'),
+        state=dict(type='str', default='query', choices=['query', 'absent', 'present']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[['state', 'absent', ['name'], ['site_name']],
+                     ['state', 'present', ['name'], ['site_name']]]
+    )
+
+    nd = NDModule(module)
+    ndi = NDI(nd)
+
+    state = nd.params.get("state")
+    name = nd.params.get("name")
+    site_name = nd.params.get('site_name')
+    insights_group = nd.params.get('insights_group')
+    description = nd.params.get('description')
+    file = nd.params.get('file')
+    manual = nd.params.get('manual')
+
+    path = 'config/insightsGroup'
+    if name is None:
+        nd.existing = ndi.query_pcvs(insights_group)
+    elif site_name is not None:
```

Comment:
what happens if name is None?

Reply:
if name is None, the module will query all the pre-change validation jobs under a insights group